rfcs/2021-05-17-7469-load-balancing-aggregators.md

```diff
@@ -0,0 +1,99 @@
+# RFC 7469 - 2021-05-16 - Scaling and load balancing for Vector aggregators
+
+This RFC describes the need for a tooling and environment agnostic load balancing solution to be bundled with Vector aggregator deployments.
+
+* [Scope](#scope)
+* [Motivation](#motivation)
+* [Internal Proposal](#internal-proposal)
+* [Rationale](#rationale)
+* [Prior Art](#prior-art)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+  * [Do Nothing](#do-nothing)
+  * [Only Client-side Load Balancing](#only-client-side-load-balancing)
+  * [Require a Service Mesh](#require-a-service-mesh)
+* [Outstanding Questions](#outstanding-questions)
+* [Plan Of Attack](#plan-of-attack)
+
+## Scope
+
+Load balancing will be a concern for any `sink` or `source` supported by Vector; some of which can use a general solution (ex: load balancing for our HTTP based `sinks`) and some of which are specific to the component (ex: Kafka or Elasticsearch `sinks`). Due to the breadth of the topic, this RFC will focus on three specific cases for load balancing while also giving consideration to future adoption for other components.
+
+* Vector agent to Vector aggregator
+* Datadog agent to Vector aggregator
+* Syslog agent to Vector aggregator
+
+## Motivation
+
+Today Vector lacks the capacity to scale horizontally (by increasing replicas) when deployed as an aggregator. This limits Vector aggregators in both reliability and performance, causing adoption concerns for users. A single aggregator will be limited in performance by the resources that can be dedicated to it, presumably with some (currently) unknown upper bounds. Vector aims to be vendor neutral, and as such we should provide the capacity to scale and load balance across Vector aggregators regardless of environment or upstream event collectors.
```

Comment:
Curious, is replicas the right term here? I guess it is? I usually think about that in database terms.

Reply:
Maybe not, definitely vocabulary brought from my time in Kuberentes where Deployments and StatefulSets are scaled out by increasing "replicas" - so it may be a Kube-centric thing.

Reply:
FWIW I understood "replicas" in context.